pkg/sql/logictest/testdata/logic_test/set_role

```diff
@@ -357,6 +357,15 @@ root  root  root  root
 statement ok
 SET ROLE testuser
 
+# Verify that RESET ALL does *not* affect role.
+statement ok
+RESET ALL
+
+query TTTT
+SELECT current_user(), current_user, session_user(), session_user
+----
+testuser  testuser  root  root
+
 query T
 SELECT user_name FROM crdb_internal.node_sessions
```

Comment:
you'll probably want to set user to testuser, as this test was meant to test usernames were reported correctly

Reply:
hm, but then it wouldn't test what we want -- the test now shows that `RESET ALL` does not change current_user

Reply:
as in, add

```
# Test node_sessions reports the correct session_user even if
# current_user changes
statement ok
SET ROLE testuser
```

after this green bit

Reply:
that's already done on line 358, and as the test shows, RESET ALL won't change the role.

Reply:
ohhhh... Shouldn't it reset the role though? Don't have pg on me

Reply:
no, it shouldn't - that's why i added this bit in r2 https://github.com/cockroachdb/cockroach/pull/75804/files#diff-c72939a9761243aba943b76184fdd0d835ec296161f19912ec74c6855658dd12R196

sorry i misunderstood before. i thought in your original comment about `role` being special you were saying that you already knew that PG doesn't reset it. (i confirmed it myself too)

Reply:
crazy!!!